src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingParameterResolver.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Globalization;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings.Path
+{
+    internal abstract class BindingParameterResolver
+    {
+        private static Collection<BindingParameterResolver> _resolvers;
+
+        static BindingParameterResolver()
+        {
+            // create the static set of built in system resolvers
+            _resolvers = new Collection<BindingParameterResolver>();
```

Comment:
We can easily open this up in the future for extensibility, but for now just keeping these interfaces internal.